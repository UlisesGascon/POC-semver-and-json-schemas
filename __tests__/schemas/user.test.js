const Ajv = require('ajv')
const addFormats = require('ajv-formats')
const fixtures = require('../../__fixtures__')
const clientCreatedSchema = require('../../src/schemas/user/created.json')
const clientDeletedSchema = require('../../src/schemas/user/deleted.json')

const ajv = new Ajv()
addFormats(ajv)

describe('clientDeleted Schema', () => {
  it('Should keep the same schema', () => {
    expect(clientDeletedSchema).toMatchSnapshot()
  })

  it('Should recognize a valid payload', () => {
    const validate = ajv.compile(clientDeletedSchema.schema)
    const valid = validate(fixtures.user.deleted)
    expect(valid).toEqual(true)
    expect(validate.errors).toEqual(null)
  })
})

describe('clientCreated Schema', () => {
  it('Should keep the same schema', () => {
    expect(clientCreatedSchema).toMatchSnapshot()
  })

  it('Should recognize a valid payload with all properties included', () => {
    const validate = ajv.compile(clientCreatedSchema.schema)
    const valid = validate(fixtures.user.created.allProperties)
    expect(valid).toEqual(true)
    expect(validate.errors).toEqual(null)
  })

  it('Should recognize a valid payload with minimal properties included', () => {
    const validate = ajv.compile(clientCreatedSchema.schema)
    const valid = validate(fixtures.user.created.minimalProperties)
    expect(valid).toEqual(true)
    expect(validate.errors).toEqual(null)
  })
})
