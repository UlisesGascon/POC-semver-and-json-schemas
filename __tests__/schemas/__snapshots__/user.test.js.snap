// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`clientCreated Schema Should keep the same schema 1`] = `
{
  "name": "userCreated",
  "schema": {
    "additionalProperties": false,
    "properties": {
      "age": {
        "description": "User age",
        "maximum": 120,
        "minimum": 0,
        "type": "integer",
      },
      "email": {
        "description": "User email",
        "format": "email",
        "type": "string",
      },
      "id": {
        "description": "Unique Id for the user",
        "format": "uuid",
        "type": "string",
      },
      "schemaVersion": {
        "description": "Metadata field to determine the json schema version used in the message",
        "maxLength": 14,
        "minLength": 5,
        "pattern": "^(?:0|[1-9]\\d*)\\.(?:0|[1-9]\\d*)\\.(?:0|[1-9]\\d*)$",
        "type": "string",
      },
      "timestamp": {
        "description": "Metadata field to determine the date of the message",
        "minimum": 0,
        "type": "integer",
      },
    },
    "required": [
      "id",
      "email",
      "schemaVersion",
      "timestamp",
    ],
    "title": "userCreated",
    "type": "object",
  },
  "schemaCompatibility": "1.x",
  "version": "1.0.0",
}
`;

exports[`clientDeleted Schema Should keep the same schema 1`] = `
{
  "name": "userDeleted",
  "schema": {
    "additionalProperties": false,
    "properties": {
      "id": {
        "description": "Unique Id for the user",
        "format": "uuid",
        "type": "string",
      },
      "schemaVersion": {
        "description": "Metadata field to determine the json schema version used in the message",
        "maxLength": 14,
        "minLength": 5,
        "pattern": "^(?:0|[1-9]\\d*)\\.(?:0|[1-9]\\d*)\\.(?:0|[1-9]\\d*)$",
        "type": "string",
      },
      "timestamp": {
        "description": "Metadata field to determine the date of the message",
        "minimum": 0,
        "type": "integer",
      },
    },
    "required": [
      "id",
      "schemaVersion",
      "timestamp",
    ],
    "title": "userDeleted",
    "type": "object",
  },
  "schemaCompatibility": "1.x",
  "version": "1.0.0",
}
`;
